### ISSUE TITLE
Some regex quantifiers don't work as expected

### ISSUE THREAD

**Issue by jmaslak** [rakudo/rakudo#2112]
## The Problem

This regex, demonstrated here, returns true when it shouldn't (I think!).

This is a golfed exampe -

`perl6 -e 'say "aaa" ~~ m/(a ** {2..*}) ** 2..*/'`

Output:
```
[0] red:jmaslak$ perl6 -e 'say "aaa" ~~ m/(a ** {2..*}) ** 2..*/'
｢aaa｣
 0 => ｢a｣
 0 => ｢aa｣
[0] red:jmaslak$
```

Note `perl6 -e 'say "aaa" ~~ m/(a ** 2..*) ** 2..*/'` does what I would expect it to do (returns false).  So does `perl6 -e 'say "aaa" ~~ m/(a ** 2..*) ** {2..*}/'`

## Expected Behavior

Expected behaviro is returning false.

## Actual Behavior

Returns a true value.

## Steps to Reproduce

`perl6 -e 'say "aaa" ~~ m/(a ** {2..*}) ** 2..*/'`

## Environment
* Operating system: Ubuntu 16.04
* Compiler version (`perl6 -v`): 2008.06 6.c

**Comment by ugexe** [comment]
It works with the `:ratchet` adverb. So probably a result of `{ }` ( See the answer [here](https://stackoverflow.com/questions/51408141/perl6-grammar-not-sure-about-some-syntax-in-an-example) )

**Comment by lizmat** [comment]
So, can we close this ticket?  Or should we work on documentation?

**Comment by jnthn** [comment]
@lizmat No, I think it's a backtracking bug, and `:ratchet` doesn't do backtracking, so "hides" the bug.

**Comment by ugexe** [comment]
This is fixed on main, so will be fixed in 2026.09 or later
